agent/requestor/src/market.rs

```diff
@@ -43,6 +59,13 @@ pub(crate) fn build_demand(
         cnts = cnts.and(constraints!["golem.node.debug.subnet" == subnet.clone(),]);
     };
 
+    if let Some(payment_platform) = payment_platform {
+        let payment_platform = payment_platform.to_string().to_uppercase();
+        log::info!("Using preferred payment platform: {}", payment_platform);
+        let cnt: String = format!("golem.com.payment.platform.{}.address", payment_platform);
+        cnts = cnts.and(constraints![cnt == "*".to_string(),]);
+    }
```

Comment:
Requestor should accept all payment platforms specified in `accounts`, not just a single one. The `payment_platform` variable is not needed at all. An example how to construct a constraint matching any of multiple platforms is given [here](https://docs.google.com/document/d/1dDQ2r2mJncgUVdWVVl1AQy3w3poDX.2XuWm-jaKM8SNk/edit?ts=5eea1ff3#heading=h.7q5y2isxdtsr).